gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/ApiService.java

```diff
@@ -170,6 +170,14 @@ ApiEntity importPathMappingsFromPage(
     Collection<String> searchIds(ExecutionContext executionContext, String query, Map<String, Object> filters, Sortable sortable)
         throws TechnicalException;
 
+    List<ApiName> findAllNames(
```

Comment:
I wonder if we shouldn't paginate it ? wdyt @phiz71 @gaetanmaisse   
I ask the question in view of what has been done lately. 
Even if we only take out the names here, it is a global search. 
We could page through the repository and request an Integer.max in the resource.

I'm not sure :) but I think it's good to ask the question 😁

Reply:
I would not do it as this endpoint will certainly be removed when we will rework on the dashboards. With an autocomplete on the frontend instead of a select with an infinite scroll

Reply:
Actually, we had the same kind of error on another screen. You can see more details here:
https://github.com/gravitee-io/gravitee-api-management/pull/2489.

The idea was to use an input to filter out APIs by name.

Reply:
Ok but here I would prefer to not update the dashboard itself as it's a resolver problem.
This is an old angular js component and I think it will be rework during the migration to angular.